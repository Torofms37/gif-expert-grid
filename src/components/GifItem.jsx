import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const GifItem = ({ title, url }) => {

  const [images, setImages] = useState([]);

  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  useEffect(() => {
    getImages();
  }, []);

  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
