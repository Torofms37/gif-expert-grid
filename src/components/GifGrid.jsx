import { useEffect, useState } from "react";
import { GifItem } from "./GifItem.jsx";
import { useFetchGifs } from "../hooks/useFetchGifs.js";

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category );

  return (
    <>
      <form>
        <h3>{category}</h3>

        <h2>
          { isLoading && ( <p>Cargando...</p> ) }
        </h2>

        <div className="card-grid">
          {images.map((image) => (
            <GifItem 
              key={image.id}
              {...image}
            />
          ))
          }
        </div>
      </form>
    </>
  );
};
