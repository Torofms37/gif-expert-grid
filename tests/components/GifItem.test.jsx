import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('testea a <GifItem/>', () => {
  const title = 'tete';
  const url ='www.google.com';


  test('debe coincidir con el snapshot', () => {
    const { container } = render( <GifItem title={title} url={url} />);
    expect( container ).toMatchSnapshot();
  });

  test('debe ostrar Ã±a imagen y url indicado', () => {
    render( <GifItem title={title} url={url} />);
    expect(screen.getByRole('img').src ).toBe( url );
  })
  
} ) 