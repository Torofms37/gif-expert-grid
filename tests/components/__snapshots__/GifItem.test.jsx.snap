// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testea a <GifItem/> debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="tete"
      src="www.google.com"
    />
    <p>
      tete
    </p>
  </div>
</div>
`;
